### ISSUE TITLE
Incorrectly parsed palette ID ## rendering a 1.16.1 world

### ISSUE THREAD

**Issue by Shadowdane** [mapcrafter/mapcrafter#374]
![mapcrafter_errors](https://user-images.githubusercontent.com/43715181/86920320-99eb2100-c0f7-11ea-876d-1da3643b8f3d.png)

Downloaded the Nightly Build version: `3.0 (2.3.1~273~gd3b2ed3)` for Win64.   At least that's what's indicated in the version.txt file.

Gives incorrect parsed palette ID errors with trying to render a 1.16.1 world.    Is the nightly build supporting 1.16.x?    The documentation seems to indicated that 1.16 is supported on the new builds?   I'm confused.

**Comment by fredkilbourn** [comment]
This looks like you're not using the updated branch with support for 1.16.

Did you pull the latest from https://github.com/mapcrafter/mapcrafter/tree/world113?

You'll have to recompile and use the latest binaries built off this branch.

**Comment by fredkilbourn** [comment]
@Shadowdane Where'd you download the build from?  It doesn't look like it has the latest world1.16 commits in it.

Does `mapcrafter -v` show `Mapcrafter version: 3.0 (v.2.3.1-273-gd3b2ed3)` and `mapcrafter --mc-version` show `1.16`?

Do you know if all the chunks in your map are in 1.16 format or earlier format or mixed?  you can run the minecraft server with the `--forceUpgrade` flag to force it to upgrade all chunks to the latest format.

My only thought if you don't have the latest build is that maybe you have chunks in your world that aren't updated to the 1.16 format and maybe the latest code can't handle older chunks?

**Comment by elamperti** [comment]
@fredkilbourn nightly builds are available here: https://mapcrafter.org/downloads/nightly - these are unstable and may have unexpected problems (that are usually ironed out before a stable version is released).

@Shadowdane the suggestion of using `--forceUpgrade` may solve it, otherwise I'd suggest you to wait for the next nightly build

**Comment by Shadowdane** [comment]
> @Shadowdane Where'd you download the build from? It doesn't look like it has the latest world1.16 commits in it.
> 
> Does `mapcrafter -v` show `Mapcrafter version: 3.0 (v.2.3.1-273-gd3b2ed3)` and `mapcrafter --mc-version` show `1.16`?
> 
> Do you know if all the chunks in your map are in 1.16 format or earlier format or mixed? you can run the minecraft server with the `--forceUpgrade` flag to force it to upgrade all chunks to the latest format.
> 
> My only thought if you don't have the latest build is that maybe you have chunks in your world that aren't updated to the 1.16 format and maybe the latest code can't handle older chunks?

Yup that's what it shows for me..

```
D:\Mapcrafter>mapcrafter.exe -v
Mapcrafter version: 3.0 (v.2.3.1-273-gd3b2ed3)

D:\Mapcrafter>mapcrafter.exe --mc-version
1.16
```

I can give the --forceUpgrade a try on our server but it's a brand new world which was started when 1.16 was released.   We didn't upgrade an older world at all.    All chunks have been created in 1.16.

[edit]
I did a --forceUpgrade on the world and did not make any difference.   Still the same incorrectly parsed palette ID errors.     I guess I'll need to wait on a newer Nightly update to resolve that issue.

**Comment by Shadowdane** [comment]
Any progress on this issue?      Haven't seen any new Nightly builds posted to Github in about a month.

**Comment by Borous** [comment]
I’ve been experiencing this exact issue too for weeks now, all worked well until the server and map where updated to 1.16.1, 

From what I’ve seen as well there hasn’t been any activity at all on mapcrafter, which is quite a shame because I used mapcrafter pretty much daily.

**Comment by elamperti** [comment]
@Borous There is activity in the repository, in fact if you landed on this issue it means you are using the build resulting from the latest changes.

I've been quite busy with other projects and my last efforts to check this issue back on July weren't successful. ~Hopefully someone else will be able to find a patch for this~. Please check with the latest build, as PR #375 may be the fix for this.

In any case, remember that nightly builds are unstable, and that most of the people contributing to it are doing so in their free time, so please be patient. We are close to having a stable release :)

**Comment by Borous** [comment]
Ah I see!, Thank you oh so much for fixing it!
I’ve been using mapcrafter on a server for several years now, and it has been extremely useful, 

and I apologize for incorrectly assuming the project hadn’t seen much activity, I’m rather new to using github and I haven’t got the slightest clue on how to navigate it,

**Comment by Shadowdane** [comment]
> @Borous There is activity in the repository, in fact if you landed on this issue it means you are using the build resulting from the latest changes.
> 
> I've been quite busy with other projects and my last efforts to check this issue back on July weren't successful. ~Hopefully someone else will be able to find a patch for this~. Please check with the latest build, as PR #375 may be the fix for this.
> 
> In any case, remember that nightly builds are unstable, and that most of the people contributing to it are doing so in their free time, so please be patient. We are close to having a stable release :)

thx so much the Nightly build fixed it this issue!     I'm able to render a 1.16.2 world.    I'm trying to shrink the size of the render though as it's huge at 12GB!
   
It's a SMP server and we have people that explore a lot and are building all over the map.    I'm getting this error when I try smaller texture sizes in my config.    If I don't put a texture size in the config it works, but then the filesize of the render is just massive.

```
2020-08-15 20:33:28 [INFO] [default] Scanning worlds...
2020-08-15 20:33:32 [INFO] [default] [1/1] Rendering map Shacknews20_Daylight ("Overworld"):
2020-08-15 20:33:32 [INFO] [default] [1.1/1.1] Rendering rotation top-right...
2020-08-15 20:33:32 [INFO] [default] Scanning required tiles...
2020-08-15 20:33:33 [INFO] [default] I will load block images from "D:\Mapcrafter\data\blocks" now
2020-08-15 20:33:33 [ERROR] [default] Unable to load block images: Block info file "D:\Mapcrafter\data\blocks\isometric_1_8.txt" does not exist!
2020-08-15 20:33:33 [ERROR] [default] Skipping remaining rotations.
[=========================================================] nan% 0/0 0.00/s
2020-08-15 20:33:33 [INFO] [default] [1.1/1.1] Rendering rotation top-right took 1 seconds.
2020-08-15 20:33:33 [INFO] [default] Rendering all worlds took 1 seconds.
2020-08-15 20:33:33 [INFO] [default] Finished.....aaand it's gone!
```

**Comment by Shadowdane** [comment]
Ok i found I apparently need to use Blockcrafter to generate different sized textures?

I can't figure out how to run that thing though, spent probably 3-4 hours trying to get it work but it just constantly fails with errors.

Is there anywhere to just download the block textures files for different sizes?

**Comment by elamperti** [comment]
@Shadowdane it wouldn't be practical to create/pack a lot of different texture sizes, you'll have to create the ones for your specific case with blockcrafter. Once `world113` is merged the new docs should be published with more info on that, but you can read about it now here: https://github.com/mapcrafter/mapcrafter/blob/world113/docs/blockcrafter.rst

Closing this issue as the original error related to palette IDs seems to be fixed. If someone has problems similar to the one on the first post (i.e. related to the palette), comment below, otherwise please open a new issue. Thank you!

**Comment by Shadowdane** [comment]
Thanks elamperti!

Yah i opened an issue on the Blockcrafter page as it's just related to that tool.   But not sure if people check that?     But anyway I messed with Python & Docker ALL day today and had no success getting Blockcrafter to work.     Just 8+ hours of error messages apon more error messages.   Would get one resolved and 10 more errors would pop up.    Had a friend who a programmer helping me.   But even he had issues but finally got the Docker thing working.   He got the Docker version working but he mentioned it was still running with no output after almost 6 hours which doesn't sound right.    The documentation says should take minutes??

https://github.com/mapcrafter/blockcrafter/issues/10

Anyway hope whatever problem I'm having here can get fixed.    Not sure what i'm doing wrong and the error messages are no help to me.
